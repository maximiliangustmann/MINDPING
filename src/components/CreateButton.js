import React from 'react'
import { NavLink } from 'react-router-dom'
import styled from 'styled-components/macro'

export default function CreateButton() {
  return (
    <StyledNavLink data-cy="create" to="/create">
      Create new review
    </StyledNavLink>
  )
}

const StyledNavLink = styled(NavLink)`
  display: flex;
  justify-content: center;
  align-items: center;
  text-decoration: none;
  border-radius: 50px;
  border: none;
  background: var(--gradient-red);
  box-shadow: var(--shadow-red-30);
  font-size: 1.375em;
  color: var(--main-white);
  font-weight: 900;
  padding: 20px;
  margin: 75px 40px 20px;
`
