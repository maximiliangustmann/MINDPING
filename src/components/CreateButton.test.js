import React from 'react'
import renderer from 'react-test-renderer'
import CreateButton from './CreateButton'
import { MemoryRouter } from 'react-router-dom'
import { render } from '@testing-library/react'

describe('CreateButton', () => {
  it('has the correct title', () => {
    const { getByText } = render(
      <MemoryRouter>
        <CreateButton />
      </MemoryRouter>
    )
    expect(getByText('Create new review')).toBeInTheDocument()
  })

  it('renders correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <CreateButton />
      </MemoryRouter>
    )
    expect(tree).toMatchSnapshot()
  })
})
