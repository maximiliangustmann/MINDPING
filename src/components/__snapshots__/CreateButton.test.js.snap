// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateButton renders correctly 1`] = `
<a
  aria-current={null}
  className="CreateButton__StyledNavLink-sc-4mhuep-0 dwLVYW"
  href="/create"
  onClick={[Function]}
>
  <button
    className="CreateButton__Button-sc-4mhuep-1 bGbYpS"
  >
    Create new review
  </button>
</a>
`;
